go/tasks/pluginmachinery/tasklog/template.go

```diff
@@ -96,67 +106,28 @@ func (s TemplateLogPlugin) GetTaskLog(podName, podUID, namespace, containerName,
 	return *o.TaskLogs[0], nil
 }
 
-func (s TemplateLogPlugin) GetTaskLogs(input Input) (Output, error) {
-	// Container IDs are prefixed with docker://, cri-o://, etc. which is stripped by fluentd before pushing to a log
-	// stream. Therefore, we must also strip the prefix.
-	containerID := input.ContainerID
-	stripDelimiter := "://"
-	if split := strings.Split(input.ContainerID, stripDelimiter); len(split) > 1 {
-		containerID = split[1]
+func (s TemplateLogPlugin) GetTaskLogs(input Input, extraTemplateVars ...TemplateVars) (Output, error) {
+	var err error
+	templateVars, err := input.ToTemplateVars()
+	if err != nil {
+		return Output{}, err
+	}
+	err = templateVars.Merge(extraTemplateVars...)
+	if err != nil {
+		return Output{}, err
 	}
 
 	taskLogs := make([]*core.TaskLog, 0, len(s.templateUris))
 	for _, templateURI := range s.templateUris {
+		var buf bytes.Buffer
+		t := template.Must(template.New("uri").Parse(templateURI))
```

Comment:
Switches to using text templating instead of regexes+replace.

Reply:
I believe you will find a benchmarking test in one of the test files that's comparing the two and finding a big discrepancy in performance (memory allocations?)... hence the decision... 
Things may have changed and we should run it again... 
Also we may make this as a flag on the log config... the templatingEngine: Default/RegEx, Gotemplate ?

Reply:
Hmm interesting. Didn’t know about this history. The regex implementation is somewhat limiting, but we can work around that I suppose. I’m fine with keeping it as-is if the difference is significant - not sure about adding more configuration.

Reply:
Since we are revamping this logic, can we introduce a way of flagging to switch over to the old way incase of backward compat issues. Haytham's suggestion to use a log config can allow to keep the old and new behavior and we can deprecate the old config and use the new method as default in new versions, but users can switch over in case they find issues and if things look ok, then we remove the deprecated config and method in next releases

Reply:
For performance reasons, I reverted to using regexes+replace.